packages/xdl/src/Versions.ts

```diff
@@ -50,7 +51,12 @@ type Versions = {
 export async function versionsAsync(): Promise<Versions> {
   const api = new ApiV2Client();
   const versionCache = new Cacher(
-    () => api.getAsync('versions/latest'),
+    () =>
+      api.getAsync(
+        'versions/latest',
+        undefined,
+        ConnectionStatus.isOffline() ? { timeout: 1 } : undefined
```

Comment:
maybe we can update the api client itself to default to `{ timeout: 1 }` when the connection is offline, so we don't have to be careful to add this to each api call.

some psuedocode for the api module:

```
options = options || {};
if (!options.hasOwnProperty('timeout') && ConnectionStatus.isOffline()) {
  options.timeout = 1;
}
```